cogs/annual_handover_and_reset.py

```diff
@@ -0,0 +1,198 @@
+"""Contains cog classes for annual handover and role reset functionality."""
+
+
+from collections.abc import Sequence
+
+__all__: Sequence[str] = ("HandoverCommandCog", "AnnualResetRolesCommandCog")
+
+import logging
+from logging import Logger
+from typing import Final
+
+import discord
+
+from db.core.models import GroupMadeMember
+from utils import AllChannelTypes, CommandChecks, TeXBotApplicationContext, TeXBotBaseCog
+
+logger: Logger = logging.getLogger("TeX-Bot")
+
+
+class HandoverCommandCog(TeXBotBaseCog):
+    """Cog class that defines the handover command."""
+
+    @discord.slash_command(  # type: ignore[no-untyped-call, misc]
+        name="handover",
+        description="Initiates the annual Discord handover procedure for new committee",
+    )
+    @CommandChecks.check_interaction_user_has_committee_role
+    @CommandChecks.check_interaction_user_in_main_guild
+    async def handover(self, ctx: TeXBotApplicationContext) -> None:
+        """
+        Definition & callback response of the "handover" command.
+
+        The "handover" command runs the relevant handover methods
+        which will perform the following actions:
+        - Give the "Committee" role to any users that have the "Committee-Elect" role
+        - Remove the "Committee-Elect" role from any user that has it
+        - Remove the permissions for the "Committee-Elect" role
+        from all channels except "#handover"
+
+        To do this TeX-Bot will need to hold a role above that of the committee role.
+        """
+        # NOTE: Shortcut accessors are placed at the top of the function, so that the exceptions they raise are displayed before any further errors may be sent
+        main_guild: discord.Guild = self.bot.main_guild
+        committee_role: discord.Role = await self.bot.committee_role
+        committee_elect_role: discord.Role = await self.bot.committee_elect_role
+
+        handover_channel: AllChannelTypes | None = discord.utils.get(
+            main_guild.channels,
+            name="Handover",
+        )
+
+        initial_response: discord.Interaction | discord.WebhookMessage = await ctx.respond(
+            ":hourglass: Running handover procedures... :hourglass:",
```

Comment:
* Message content should be passed to `ctx.respond()` function as a named argument: `content="...",`
* Missing `ephemeral=True,` argument

Reply:
We only use content= for when we're editing the message - we've never used it anywhere else for sending the initial message

Reply:
I'd like to start using it, to make it more explicit which argument is which. I know it may be unlikely, but if the Pycord library ever changes the order of the arguments, we don't want to end up sending the wrong value to the wrong argument.